wire-runtime/src/main/java/com/squareup/wire/ProtoAdapter.kt

```diff
@@ -46,7 +46,7 @@ abstract class ProtoAdapter<E>(
   internal var repeatedAdapter: ProtoAdapter<List<E>>? = null
 
   /** Returns the redacted form of `value`. */
-  open fun redact(value: E): E? = null
+  open fun redact(value: E): E = throw UnsupportedOperationException()
```

Comment:
Do you just want to make this `abstract`?

Reply:
Yep, good point

Reply:
That breaks binary compat though.. I'm gonna merge as is, can revisit later

Reply:
It's a major version bump though. Generated code is only guaranteed to work with the same compiler version.

Reply:
Well yeah, guess that's fine.